gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/CheckPegasusSnapshotTask.java

```diff
@@ -1,32 +1,176 @@
 /*
-   Copyright (c) 2020 LinkedIn Corp.
-
-   Licensed under the Apache License, Version 2.0 (the "License");
-   you may not use this file except in compliance with the License.
-   You may obtain a copy of the License at
-
-       http://www.apache.org/licenses/LICENSE-2.0
-
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
-*/
+ * Copyright (c) 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.linkedin.pegasus.gradle.tasks;
 
+
+import com.linkedin.pegasus.gradle.PathingJarUtil;
+import com.linkedin.pegasus.gradle.internal.CompatibilityLogChecker;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
 import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.FileCollection;
 import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.Classpath;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.SkipWhenEmpty;
 import org.gradle.api.tasks.TaskAction;
 
 
 @CacheableTask
 public class CheckPegasusSnapshotTask extends DefaultTask
 {
+  private File _currentSnapshotDirectory;
+  private File _previousSnapshotDirectory;
+  private String _compatibilityLevel;
+  private FileCollection _codegenClasspath;
+  private String _compatibilityMode;
+  private FileCollection _handlerJarPath;
+  private boolean _isExtensionSchema = false;
```

Comment:
since we might use pdl for idl's in 3.0 or for any other future usecase, recommend using 'FileCompatibilityType' instead of boolean flag.

Reply:
I don't think using Enum instead of a boolean is appropriate here. No matter how many use cases of pdl in the future, it is unrelated to the usage of this boolean flag here. isExtensionSchema boolean not only is used to detect input but also is used for annotation check in the following PR.